File: client/src/app/_models/part.ts
import { Requisition } from "./requisiton";
import { Supplier } from "./supplier";

export interface Part {
    id: number,
    partCode: string,
    description: string,
    notes: string,
    bufferValue: number,
    bufferUnit: string;
    buffer: string,
    supplySources: SupplySource[]
    requisitions: Requisition[];
}

export interface SupplySource {
    id: number,
    partId: number,
    supplier: Supplier,
    supplierSKU: string,
    manufacturerSKU: string,
    packSize: number,
    minimumOrderQuantity: number,
    notes: string,
    roHS: boolean,
    prices: SourcePrice[]
}

export interface SourcePrice {
    id: number,
    unitPrice: number,
    quantity: number,
    priceString: string
}

File: API/Entities/Part.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace API.Entities
{
    public class Part
    {
        public int Id { get; set; }

        [Required]
        public string PartCode { get; set; }
        public string Description { get; set; }
        public string Notes { get; set; }
        public float BufferValue { get; set; }
        public string BufferUnit { get; set; }
        public ICollection<SupplySource> SupplySources { get; set; }

        public string Buffer { get { return $"{BufferValue} {BufferUnit}"; } }


        public ICollection<OutboundOrderItem> Orders { get; set; }
        public ICollection<Requisition> Requisitions { get; set; }
    }
}